spec/support/rbac_shared_contexts.rb

```diff
@@ -0,0 +1,41 @@
+RSpec.shared_context "rbac_objects" do
+  let(:app_name) { 'catalog' }
+  let(:resource) { "portfolios" }
+  let(:verbs) { ["read"] }
+  let(:resource_id1) { "1" }
+  let(:resource_id2) { "2" }
+  let(:resource_id3) { "3" }
+  let(:group1) { double(:name => 'group1', :uuid => "123") }
+  let(:group2) { double(:name => 'group2', :uuid => "12345") }
+  let(:group3) { double(:name => 'group3', :uuid => "45665") }
+  let(:role1) { double(:name => 'role1-Sharing', :uuid => "67899") }
+  let(:role2) { double(:name => 'role2', :uuid => "56779") }
+  let(:role1_detail) { double(:name => role1.name, :uuid => role1.uuid, :access => [access1]) }
+  let(:role2_detail) { double(:name => role2.name, :uuid => role2.uuid, :access => []) }
+  let(:role1_detail_updated) { double(:name => role1.name, :uuid => role1.uuid, :access => []) }
+  let(:groups) { [group1, group2, group3] }
+  let(:roles) { [role1] }
+  let(:policy1) { double(:group => group1, :roles => roles) }
+  let(:policies) { [policy1] }
+  let(:attr_filter1) { double(:value => resource_id1) }
+  let(:attr_filter2) { double(:value => resource_id2) }
+  let(:attr_filter3) { double(:value => resource_id3) }
+  let(:resource_def1) { double(:attribute_filter => attr_filter1) }
+  let(:resource_def2) { double(:attribute_filter => attr_filter2) }
+  let(:resource_def3) { double(:attribute_filter => attr_filter3) }
```

Comment:
These can all be condensed since the `attr_filter#`s aren't being used anywhere else, I don't think there's too much value in having them be their own `let`s:

`let(:resource_def1) { double(:attribute_filter => double(:value => resource_id1)) }`